pandas/io/tests/test_pickle.py

```diff
@@ -284,6 +284,46 @@ def test_pickle_v0_15_2(self):
         #
         tm.assert_categorical_equal(cat, pd.read_pickle(pickle_path))
 
+    def compression_explicit(self, compression):
+        # issue 11666
+        if compression == 'xz':
+            tm._skip_if_no_lzma()
+        with tm.ensure_clean(self.path) as path:
+            df = tm.makeDataFrame()
+            df.to_pickle(path, compression=compression)
+            df2 = pd.read_pickle(path, compression=compression)
+            tm.assert_frame_equal(df, df2)
+
+    def test_compression_explicit(self):
+        compressions = [None, 'gzip', 'bz2', 'xz']
+        for c in compressions:
+            yield self.compression_explicit, c
+
+    def compression_explicit_bad(self, compression):
```

Comment:
I'm not sure it makes sense to test for invalid `compression` values or for compression inference. This is because you use `_infer_compression`, which gets tested for `read_table`. It seems repetitive to have every `read` method have to test whether compression inference now that we've consolidated things. Interested whether others agree?